File: Algorithms/Implementation/Easy/Designer PDF Viewer/pdf.cpp
#include <iostream>
#include <vector>

int designerPdfViewer(std::vector<int> h, std::string word) {
	int highest = 0;

	for (int i = 0; i < word.size(); i++) {
		if (h[word[i] - 97] > highest) {
			highest = h[word[i] - 97];
		}
	}

	return highest * word.size();
}

int main()
{
	std::vector<int> heights(26, 0);
	for (int i = 0; i < 26; i++) {
		std::cin >> heights[i];
	}

	std::string word = "";
	std::cin >> word;

	std::cout << designerPdfViewer(heights, word);
}
	

File: Algorithms/Implementation/Easy/Designer PDF Viewer/pdf.js
'use strict';

process.stdin.setEncoding('utf-8');
var os = require('os');
var input = "";

process.stdin.on("data", data => {
	input += data;

	if (data.split(os.EOL)[0] == "") {
		process.stdin.emit('end');
	}
});

process.stdin.on('end', _ => {
	main();
});

function main() {
	var heights = input.split(os.EOL)[0].split(" ").map(x=>+x);
	var word = input.split(os.EOL)[1];
	console.log(designerPdfViewer(heights, word));
}

function designerPdfViewer(h, word) {
	var highest = 0;

	for (var w of word) {
		if (h[w.charCodeAt(0) - 97] > highest) {
			highest = h[w.charCodeAt(0) - 97];
		}
	}

	return highest * word.length;
}